clio-transfer-model/src/main/scala/org/broadinstitute/clio/transfer/model/bam/BamExtensions.scala

```diff
@@ -1,20 +1,25 @@
 package org.broadinstitute.clio.transfer.model.bam
 
+import java.net.URI
+
 object BamExtensions {
 
   /** File extension for bams. */
   val BamExtension = ".bam"
 
   /** Extra extension for bam indices. */
-  val BaiExtensionAddition = ".bai"
-
-  /** Full extension for bam indices. */
-  val BaiExtension = s"$BamExtension$BaiExtensionAddition"
+  val BaiExtension = ".bai"
 
   /** Extra extension for bam md5s. */
   val Md5ExtensionAddition = ".md5"
 
   /** Full extension for bam md5s. */
   val Md5Extension = s"$BamExtension$Md5ExtensionAddition"
 
+  def replaceBamExtensionWithBaiExtension(bamUri: URI): URI = {
+    URI.create(
+      s"${bamUri.toString.replace(BamExtensions.BamExtension, BamExtensions.BaiExtension)}"
```

Comment:
any chance of this replacing a mid-string `.bam`, if there's a sample alias containing that?

Reply:
There is definitely that chance. I really hope that nobody _does_ this, but that can't be guaranteed.